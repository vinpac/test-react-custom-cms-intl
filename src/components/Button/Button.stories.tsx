import React from 'react'
import { storiesOf } from '@storybook/react'

import { withInfo } from '@storybook/addon-info'
import Button from '~/components/Button'
const text = `
A very important part of this example is the \`.babelrc\` file which configures the test
environment to use \`babel-preset-env\` and configures it to transpile modules to commonjs).
Learn more.

Hello world
`

storiesOf('Button', module)
  .add(
    'Default',
    withInfo({ text })(() => (
      <Button color="blue" position={{ x: 1, y: 2 }} text="qw" />
    )),
  )
  .add(
    'Loading',
    withInfo({ text })(() => (
      <Button color="green" position={{ x: 1, y: 2 }} text="qw" />
    )),
  )
