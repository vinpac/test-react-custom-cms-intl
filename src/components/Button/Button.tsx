import * as React from 'react'

interface ButtonProps {
  /** Buttons background color */
  color?: 'blue' | 'green'
  /** Buttons background color
   * @default "qwe"
   */
  text: string
  position: {
    x: number
    y: number
  }
}

const Button: React.SFC<ButtonProps> = props => (
  <div className="btn btn-primary">{props.color} = Home</div>
)

export default Button
