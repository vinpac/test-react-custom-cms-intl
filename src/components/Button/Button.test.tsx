/* eslint-env jest */
import React from 'react'
import { create } from 'react-test-renderer'
import Button from '~/components/Button'

describe('NiceCheckbox', () => {
  it('renders the checkbox with correct label', () => {
    expect(
      create(
        <Button text="qwe" position={{ x: 1, y: 2 }} color="blue" />,
      ).toJSON(),
    ).toMatchSnapshot()
    expect(
      create(<Button text="qwe" position={{ x: 1, y: 2 }} />).toJSON(),
    ).toMatchSnapshot()
  })
})
