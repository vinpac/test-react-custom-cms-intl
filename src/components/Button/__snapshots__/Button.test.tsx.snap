// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NiceCheckbox renders the checkbox with correct label 1`] = `
<div
  className="btn btn-primary"
>
  blue
   = Home
</div>
`;

exports[`NiceCheckbox renders the checkbox with correct label 2`] = `
<div
  className="btn btn-primary"
>
   = Home
</div>
`;
